pkg/controller/nodeipam/ipam/cidr_allocator.go

```diff
@@ -116,6 +132,12 @@ func New(kubeClient clientset.Interface, cloud cloudprovider.Interface, nodeInfo
 	switch allocatorType {
 	case RangeAllocatorType:
 		return NewCIDRRangeAllocator(kubeClient, nodeInformer, allocatorParams, nodeList)
+	case MultiCIDRRangeAllocatorType:
+		cccList, err := listClusterCIDRs(kubeClient)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
why do we have to do this list here?
should not be part of `NewMultiCIDRRangeAllocator` ?

Reply:
it seems that at the beginning it will only check if the new API is enabled, because the list will be always empty

Reply:
I think that is better if we can move this api calls inside the constructor

Reply:
It would help collect the surface area together in more cohesive place.

Reply:
... but this `listClusterCIDRs` will only fail if the new API is not enabled, and this new API is only used by `MultiCIDRRangeAllocatorType` ... it is not a big deal, but I think that it will be more testable inside `NewMultiCIDRRangeAllocator`

Reply:
so, this get's stuck with 
```
E0802 12:38:28.473088 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
E0802 12:38:30.474714 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
E0802 12:38:33.676070 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
E0802 12:38:38.797613 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
E0802 12:38:46.990669 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
E0802 12:39:00.099762 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
E0802 12:39:21.072002 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
E0802 12:39:54.627631 1941392 cidr_allocator.go:185] Failed to list all clusterCIDRs: the server could not find the requested resource
```

but since this is feature gated, this can not happen, right?

Reply:
once the feature gate is enabled this can not fail, maybe is that I can't understand what this tries to achieve

Reply:
(Note: my comment is +1 to your suggestion).

Reply:
oh, :)